engine/cache/mountsync.go

```diff
@@ -91,11 +91,23 @@ func (m *manager) StartCacheMountSynchronization(ctx context.Context) error {
 
 	m.stopCacheMountSync = func(ctx context.Context) error {
 		var eg errgroup.Group
+
+		allCacheMounts := map[string]struct{}{}
+		core.SeenCacheKeys.Range(func(k any, v any) bool {
+			allCacheMounts[k.(string)] = struct{}{}
+			return true
+		})
+
 		for _, syncedCacheMount := range syncedCacheMounts {
 			syncedCacheMount := syncedCacheMount
```

Comment:
nit: copy not needed anymore since no go routine

Reply:
true, forgot to remove that one. fixed